### PR TITLE
feat: SendGrid dynamic templates

### PR DESCRIPTION
**Pull request by shirleyodeng**
## Summary

This PR uses SendGrid dynamic templates to send emails [noted here](https://www.notion.so/talentprotocol/New-Transactional-Emails-text-based-75f65983aebf4278a3859bf4f9b313c6).

## Notion link

https://www.notion.so/talentprotocol/Bounty-Setup-Sendgrid-e260a94d1f8f408f9f5b549d27cafe70

### CODE REVIEW

File: app/mailers/application_mailer.rb [hunk 1]
```diff
@@ -4,6 +4,31 @@ class ApplicationMailer < ActionMailer::Base
...
+  end
+
+  def sendgrid_request_body(dynamic_template_data:, template_id:, to:)
+    {
+      dynamic_template_data:,
+      from: {
+        email: "no-reply@talentprotocol.com",
+        name: "Talent Protocol"
+      },
+      mail_settings: {sandbox_mode: {enable: !Rails.env.production?}},
```
Comment (shirleyodeng) [inline app/mailers/application_mailer.rb:26]: Can't find ways to test this via the API except this: https://docs.sendgrid.com/for-developers/sending-email/sandbox-mode

Comment (RubenSousaDinis) [reply app/mailers/application_mailer.rb:26]: Looks like the correct way for me!

File: app/mailers/application_mailer.rb [hunk 2]
```diff
@@ -4,6 +4,31 @@ class ApplicationMailer < ActionMailer::Base
...
+  def sendgrid_api
+    @sendgrid_api ||= SendGrid::API.new(api_key: ENV["SENDGRID_API_KEY"])
+  end
+
+  def sendgrid_request_body(dynamic_template_data:, template_id:, to:)
+    {
+      dynamic_template_data:,
+      from: {
+        email: "no-reply@talentprotocol.com",
+        name: "Filipe at Talent Protocol"
```
Comment (RubenSousaDinis) [inline app/mailers/application_mailer.rb:24]: I've created this env var so we can distinguish emails from prod and staging. I've added in both environments. Can you add it o the `.env` file?

```suggestion
        name: ENV['EMAILS_FROM']
```

Comment (shirleyodeng) [reply app/mailers/application_mailer.rb:24]: Okay sounds good!

File: app/mailers/application_mailer.rb [hunk 3]
```diff
@@ -4,6 +4,31 @@ class ApplicationMailer < ActionMailer::Base
 
   private
 
+  def send_sendgrid_email(dynamic_template_data:, template_id:, to:)
+    sendgrid_api
```
Comment (shirleyodeng) [inline app/mailers/application_mailer.rb:8]: @RubenSousaDinis if the response is not successful, do you just want me to raise an exception with the response body?

**Comment by RubenSousaDinis** [comment]
I think we can remove all email templates from the app in this pull request, no? I only see some of them being deleted

File: app/mailers/application_mailer.rb (hunk 1 above)
Comment (shirleyodeng) [reply app/mailers/application_mailer.rb:26]: Okay thanks, just can't seem to test this locally

**Comment by shirleyodeng** [comment]
> I think we can remove all email templates from the app in this pull request, no? I only see some of them being deleted

@RubenSousaDinis not all emails are being transferred to SendGrid

File: app/mailers/application_mailer.rb (hunk 1 above)
Comment (RubenSousaDinis) [reply app/mailers/application_mailer.rb:26]: It doesn't trigger the http request? In the docs they mention it

Comment (shirleyodeng) [reply app/mailers/application_mailer.rb:26]: It does but it just checks the arguments and doesn't test the email eg. I can't see what it looks like

**Pull request closed without merging**